### ISSUE TITLE
Elements in iframes unreachable

### ISSUE THREAD

**Issue by MarioStoilov** [SeleniumHQ/selenium#2235]
## Meta -
OS:  Windows 10
Selenium Version:  
2.53
Browser:  
Firefox 
Browser Version:  47 (update 8 june, dunno the exact version)

## Steps to reproduce -
Request a dev instance here -> http://www.servicenow.com/
run this code (ruby):
```
@driver = Selenium::WebDriver.for :firefox, marionette: true
@driver.manage.timeouts.implicit_wait = 30
@accept_next_alert = true
BASE_URL = *your instance url here*
@driver.get(BASE_URL + '/nav_to.do?uri=%2Fchange_request.do%3Fsysparm_stack%3Dchange_request_list.do%26sys_id%3D-1%26sysparm_query%3Dactive%3Dtrue')
@driver.switch_to().frame('gsft_main')
@driver.find_element(:id, 'change_request.form_scroll') #failure
```
Error is:
```
1465379820627	Marionette	TRACE	conn0 -> [0,48,"findElement",{"using":"css selector","value":"#change_request.form_scroll"}]
JavaScript error: chrome://marionette/content/proxy.js, line 228: TypeError: l is undefined
1465379850654	Marionette	TRACE	conn0 <- [1,48,{"error":"no such element","message":"Unable to locate element: #change_request.form_scroll","stacktrace":null},null]

Selenium::WebDriver::Error::NoSuchElementError: Unable to locate element: #change_request.form_scroll
```

**Comment by davehunt** [comment]
> as per your unwillingness to fix a simple bug, which prevents firefox starting using the old way

This is not an accurate or useful comment. There may be a point release of Firefox 47 that restores support for Selenium next week, however there is still a recommendation to move to Marionette. A future version of Firefox (currently now scheduled for 48) will block unsigned add-ons, preventing the ageing implementation of the Firefox driver from working.

Marionette is the replacement for the ageing Firefox driver, developed and maintained by Mozilla. This will mean much better support going forward, but unfortunately does mean there are some use features not fully supported yet. We need users to start switching to Marionette to help us to identify these missing features. Bugs for Marionette can be raised [here](https://bugzilla.mozilla.org/buglist.cgi?product=Testing&component=Marionette), and for GeckoDriver [here](https://github.com/mozilla/geckodriver/issues).

**Comment by lukeis** [comment]
If this is a marionette bug, which appears to be. Please log the issue where appropriate with the links Dave provided. If the browser launches and you are able to perform any commands first, the issue will lie with Mozilla.

**Comment by titusfortner** [comment]
@MarioStoilov - 
> (as per your unwillingness to fix a simple bug, which prevents firefox starting using the old way)

Legacy Firefox is supported in 2.53.0. What issue do you have?

Also, I cherry-picked @p0deje's selctor escape commit into the leg-rc branch and released selenium-webdriver-2.53.1 which should solve this issue.
